src/smc-webapp/course/common.cjsx

```diff
@@ -365,3 +368,180 @@ exports.StudentAssignmentInfo = rclass
+                <Button onClick={@clear_and_focus_search_input}>
+                    Cancel
+                </Button>
+            </ButtonToolbar>
+        </div>
+
+    render_add_selector_button : ->
+        num_items_selected = @state.selected_items?.length ? 0
+        btn_text = switch @props.search_results.length
+            when 0 then "No #{@props.item_name} found"
+            when 1 then "Add #{@props.item_name}"
+            else switch num_items_selected
+                when 0 then "Select #{@props.item_name} above"
+                when 1 then "Add selected #{@props.item_name}"
+                else "Add #{num_items_selected} #{@props.item_name}s"
+        disabled = @props.search_results.length == 0 or (@props.search_results.length >= 2 and num_items_selected == 0)
+        <Button disabled={disabled} onClick={@add_button_clicked}><Icon name="plus" /> {btn_text}</Button>
+
+    render : ->
+        <div>
+            <SearchInput
+                autoFocus     = {true}
+                ref           = 'search_input'
+                default_value = ''
+                placeholder   = "Add #{@props.item_name} by folder name (enter to see available folders)..."
+                on_submit     = {@props.do_search}
+                on_escape     = {@clear_and_focus_search_input}
+                buttonAfter   = {@search_button()}
+            />
+            {@render_add_selector() if @state.show_selector}
+         </div>
+
+# Definitely not a good abstraction.
+# Purely for code reuse (bad reason..)
+# Complects FilterSearchBar and AddSearchBar...
+exports.FoldersToolbar = rclass
+    propTypes :
+        search        : rtypes.string
+        search_change : rtypes.func.isRequired      # search_change(current_search_value)
+        num_omitted   : rtypes.number
+        project_id    : rtypes.string
+        items         : rtypes.object.isRequired
+        add_folders   : rtypes.func                 # add_folders (Iterable<T>)
+        item_name     : rtypes.string
+        plural_item_name : rtypes.string
+
+    getDefaultProps : ->
+        item_name : "item"
+        plural_item_name : "items"
+
+    getInitialState : ->
+        add_is_searching : false
+
+    do_add_search : (search) ->
+        if @state.add_is_searching
+            return
+        @setState(add_is_searching:true)
+        salvus_client.find_directories
+            project_id : @props.project_id
+            query      : "*#{search.trim()}*"
+            cb         : (err, resp) =>
+                if err
+                    @setState(add_is_searching:false, err:err, add_search_results:undefined)
+                else
+                    filtered_results = @filter_results(resp.directories, search, @props.items)
+                    @setState(add_is_searching:false, add_search_results:filtered_results)
+
+    # Filter directories based on contents of all_items
+    filter_results : (directories, search, all_items) ->
```

Comment:
Partial fix to #67 here in the following way:

1. Filter out items that are added and *not* deleted. Deleted items will show up in add search.
1. Matches paths exactly for omitting them. ie. if `assignments/` is deleted, `assignments/01` still shows up
2. However, nothing has changed if the deleted item is added..

Okay crap. 3. means that this is probably no good and needs to be changed before merging. My planned solution is to keep 1 and to only undelete deleted items when they get added again. This "fixes" the case where an instructor gets confused when a folder they have isn't active and can't be found (but is actually in the deleted area).

Obviously can be re-implemented to trash/deletion later if desired.